src/utils/filterPeople.ts

```diff
@@ -0,0 +1,39 @@
+import { GenderFilters, Person, SearchQueries } from '../types';
+
+export function filterPeople(
+  people: Person[],
+  params: URLSearchParams,
+): Person[] {
+  let filtered = [...people];
+
+  if (params.has(SearchQueries.sex)) {
+    if (params.get(SearchQueries.sex) !== GenderFilters.All) {
+      filtered = filtered.filter(
+        person => person.sex === params.get(SearchQueries.sex),
+      );
+    }
+  }
+
+  if (params.has(SearchQueries.centuries)) {
+    filtered = filtered.filter(person => {
+      return params.getAll(SearchQueries.centuries).some(centr => {
+        const personCentr = Math.floor(+person.born / 100 + 1);
+
+        return personCentr === +centr;
+      });
+    });
+  }
```

Comment:
Similar to the previous comment, you are calling `params.getAll(SearchQueries.centuries)` inside the filter function, which means it gets called for each person. It would be more efficient to call it once before the filter and store the result in a variable.